### PR TITLE
Documentation: Updated Light Docs

### PR DESCRIPTION
**Pull request by Mugen87**
This PR updates the light documentation. It adds docs for `LightShadow`, `SpotLightShadow` and `DirectionalLightShadow` and cleans up existing files. Besides, i avoid to document properties and methods multiple times. Instead i refer to base classes like `Light` or `LightShadow`.

### CODE REVIEW

**Comment by mrdoob** [comment]
Many thanks!

**Comment by Mugen87** [comment]
In addition, i would just like to make a clarification: three.js... is... cool! Period!

**Comment by mrdoob** [comment]
Awww! 😊

File: docs/api/lights/SpotLight.html [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<!DOCTYPE html>
+	<!DOCTYPE html>
```
Comment (mrdoob) [inline docs/api/lights/SpotLight.html]: Woops!

Comment (Mugen87) [inline docs/api/lights/SpotLight.html]: Sry. I will correct this...

**Pull request merged**